src/routers/handlers/check_details/index.ts

```diff
@@ -45,6 +45,10 @@ export class CheckDetailsHandler extends GenericHandler<CheckDetailsViewData> {
         logger.info(`CheckDetailsHandler execute called`);
         const viewData = this.getViewData(req);
 
+        if (req.query.lang){
+            setSessionLanguage(selectLang(req.query.lang), req);
+        }
```

Comment:
Is this not done in the middleware?

Reply:
Yes it is for some reason doesn't work appropriately 
will have a look at why that happens.